libglusterfs/src/common-utils.c

```diff
@@ -3063,6 +3063,12 @@ gf_set_volfile_server_common(cmd_args_t *cmd_args, const char *host,
     }
 
     char *lastptr = rindex(duphost, ':');
+
+    /* Handle IPV6 IP addr without port number */
+    if (lastptr && strlen(lastptr) > 1 && *(lastptr - 1) == ':') {
```

Comment:
I think there are two issues here:

1. If the input is just ":1234", for example, `*(lastptr -1)` will access undefined memory.
2. Input "::1234" is ambiguous. It may represent the address "0:0:0:0:0:0:0:1234" (without port) or the address "::" with port 1234.

I've found this: https://www.rfc-editor.org/rfc/rfc5952#section-6. So maybe we should decide what formats to accept before we get into backward compatibility issues in the future (or not make it worse if we are already there).